### PR TITLE
hareThirdParty.hare-png: init at unstable-2023-09-09

### PR DESCRIPTION
**Pull request by starzation**
## Description of changes

[hare-png](https://git.sr.ht/~sircmpwn/hare-png) package provides an implementation of the PNG image format.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by starzation** [comment]
Required dependencies from this pr: #272000.

File: pkgs/development/hare-third-party/hare-png/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  src = fetchFromSourcehut {
+    owner = "~sircmpwn";
+    repo = "hare-png";
+    rev = "9deb848ae691d784c5cc7949153cfa04c67f90b6";
+    hash = "sha256-Q7xylsLVd/sp57kv6WzC7QHGN1xOsm7YEsYCbY/zi1Q=";
+  };
+
+  nativeBuildInputs = [ hare ];
+  buildInputs = [ hareThirdParty.hare-compress ];
```
Comment (onemoresuza) [inline pkgs/development/hare-third-party/hare-png/default.nix:15]:
```suggestion
  propagatedBuildInputs = [ hareThirdParty.hare-compress ];
```

File: pkgs/development/hare-third-party/hare-png/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    export HARECACHE="$NIX_BUILD_TOP/.harecache"
+    export BINOUT="$NIX_BUILD_TOP/.bin"
+
+    makeFlagsArray+=(
+      PREFIX="${builtins.placeholder "out"}"
+    )
+
+    runHook postConfigure
+  '';
```
Comment (onemoresuza) [inline pkgs/development/hare-third-party/hare-png/default.nix:28]: The `configurePhase` may be simplified into the `makeFlags` array. `BINOUT` is variable specific to build `Hare` itself.

```suggestion
  makeFlags = [
    "PREFIX=${builtins.placeholder "out"}"
    "HARECACHE=.harecache"
  ];
```

File: pkgs/development/hare-third-party/hare-png/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  doCheck = true;
+
+  meta = with lib; {
+    homepage = "https://git.sr.ht/~sircmpwn/hare-png/";
+    description = "PNG implementation for Hare";
+    license = with licenses; [ mpl20 ];
+    maintainers = with maintainers; [ starzation ];
+
+    inherit (harec.meta) platforms badPlatforms;
```
Comment (onemoresuza) [inline pkgs/development/hare-third-party/hare-png/default.nix:38]: The same may be achieved through `hare`, removing the need of `harec` as a pattern.

```suggestion
    inherit (hare.meta) platforms badPlatforms;
```

File: pkgs/development/hare-third-party/hare-png/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, hare, harec, hareThirdParty, fetchFromSourcehut }:
```
Comment (onemoresuza) [inline pkgs/development/hare-third-party/hare-png/default.nix:1]:
```suggestion
{ lib, stdenv, hare, hareThirdParty, fetchFromSourcehut }:
```

File: pkgs/top-level/hare-third-party.nix
Comment (onemoresuza) [inline pkgs/top-level/hare-third-party.nix:1]: There seems to be a merge conflict regarding this file. A rebase should solve it.

**Review by onemoresuza — CHANGES_REQUESTED** [review]

**Comment by starzation** [comment]
@ofborg build hareThirdParty.hare-png

**Comment by starzation** [comment]
Look like dns that ofborg server use doesn't propagated

Here logs:
[x86 (`ofborg-evaluator-4`)](https://logs.ofborg.org/?key=nixos/nixpkgs.272001&attempt_id=a1d09d5c-afa5-4811-b7b1-f3d5cebef142)
[arm (`grahamc-aarch64-community-16`)](https://logs.ofborg.org/?key=nixos/nixpkgs.272001&attempt_id=3705f9ad-f5d6-4e81-84ba-673b2637eda0)

Also: [sourcehut outage](https://outage.sr.ht/)

**Comment by starzation** [comment]
@ofborg build hareThirdParty.hare-png

**Comment by onemoresuza** [comment]
# Reviewed points

Result of `nixpkgs-review pr 272001` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hareThirdParty.hare-png</li>
  </ul>
</details>

- [X] package path fits guidelines
- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package builds on x86_64-linux
- [ ] executables tested on x86_64-linux
	- N/A: Package has no executables
- [X] `meta.description` is set and fits guidelines
- [X] `meta.license` fits upstream license
- [X] `meta.platforms` is set
- [X] `meta.maintainers` is set
- [X] build time only dependencies are declared in `nativeBuildInputs`
- [X] source is fetched using the appropriate function
- [X] the list of `phases` is not overridden
- [ ] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
	- N/A: No phase is overridden
- [ ] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
	- N/A: No patches are applied 
- [ ] patches that are remotely available are fetched rather than vendored
	- N/A: No patches are applied

**Review by onemoresuza — APPROVED** [review]

**Comment by pbsds** [comment]
@ofborg build hareThirdParty.hare-png

**Pull request merged**